### PR TITLE
Coverage info

### PR DESCRIPTION
**Pull request by winsvega**
rebuild ethereum into cpp-ethereum/build with -DPROFILING=1 flag
and run ./getcoverage.sh from the main cpp-ethereum directory

### CODE REVIEW

File: getcoverage.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+if [ ! -d "$CPP_ETHEREUM_PATH/build/test" ]; then
+   echo "You need to compile and build ethereum with cmake -DPROFILING option to the build dir!"
+   exit;	 
+fi
+
+if which lcov >/dev/null; then
+    if which genhtml >/dev/null; then
+	echo Running testeth...
+	$($CPP_ETHEREUM_PATH/build/test/testeth)
```
Comment (chfast) [inline getcoverage.sh]: You need to clear capture data before running the test. `lcov --zerocounters`.

File: getcoverage.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    fi
+else
+    echo lcov not found
+    exit;
+fi
+
+OUTPUT_DIR="$CPP_ETHEREUM_PATH/build/test/coverage"
+
+TESTETH=$CPP_ETHEREUM_PATH/build/test/CMakeFiles/testeth.dir
+lcov --capture --directory $TESTETH --output-file $OUTPUT_DIR/coverage.info
```
Comment (chfast) [inline getcoverage.sh]: After capturing the data you should extract only cpp-ethereum related data. `lcov --extract */cpp-ethereum/*`.

File: test/libethereum/state.cpp [hunk 3]
```diff
@@ -129,9 +129,9 @@ BOOST_AUTO_TEST_CASE(stPreCompiledContracts)
 	dev::test::executeTests("stPreCompiledContracts", "/StateTests",dev::test::getFolder(__FILE__) + "/StateTestsFiller", dev::test::doStateTests);
 }
 
-BOOST_AUTO_TEST_CASE(stPrecompiledContractsTransaction)
+BOOST_AUTO_TEST_CASE(stPreCompiledContractsTransaction)
```
Comment (chfast) [inline test/libethereum/state.cpp]: Does not seem to be related.

Comment (winsvega) [inline test/libethereum/state.cpp]: just a little rename

File: getcoverage.sh [hunk 4]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+CPP_ETHEREUM_PATH=$(pwd)
+
+if [ ! -d "$CPP_ETHEREUM_PATH/build/test" ]; then
```
Comment (chfast) [inline getcoverage.sh]: It works only if you have build in "build" dir. Instead of `CPP_ETHEREUM_PATH` point to build dir (e.g. `CPP_ETHEREUM_BUILD_DIR`)

File: getcoverage.sh [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+   echo "You need to compile and build ethereum with cmake -DPROFILING option to the build dir!"
+   exit;	 
+fi
+
+OUTPUT_DIR="$CPP_ETHEREUM_PATH/build/test/coverage"
+TESTETH=$CPP_ETHEREUM_PATH/build  #/test/CMakeFiles/testeth.dir
```
Comment (chfast) [inline getcoverage.sh]: Wrong variable name.

File: getcoverage.sh [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+	echo genhtml not found
+	exit;
+    fi
+else
+    echo lcov not found
+    exit;
+fi
+
+echo Cleaning previous report...
+rm -r $OUTPUT_DIR/testeth
```
Comment (chfast) [inline getcoverage.sh]: Please ignore error messages from `rm` commands

File: getcoverage.sh [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+   echo "You need to compile and build ethereum with cmake -DPROFILING option to the build dir!"
+   exit;	 
+fi
+
+OUTPUT_DIR="$CPP_ETHEREUM_PATH/build/test/coverage"
+TESTETH=$CPP_ETHEREUM_PATH/build  #/test/CMakeFiles/testeth.dir
+
+if which lcov >/dev/null; then
+    if which genhtml >/dev/null; then
+	lcov --directory $TESTETH --zerocounters
```
Comment (chfast) [inline getcoverage.sh]: This file uses both spaces and tabs for formatting.

File: getcoverage.sh [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+	echo Running testeth...
+	$($CPP_ETHEREUM_PATH/build/test/testeth)
```
Comment (chfast) [inline getcoverage.sh]: Why not `$CPP_ETHEREUM_PATH/build/test/testeth`?

**Comment by CJentzsch** [comment]
Since it takes a while to run --all tests with standard VM and jit, it would be nice to make the --all optional for the bash script.

File: getcoverage.sh [hunk 9]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+CPP_ETHEREUM_PATH=$(pwd)
+
+which $CPP_ETHEREUM_PATH/build/test/testeth >/dev/null 2>&1
```
Comment (chfast) [inline getcoverage.sh]: Please remove "build" build dir assumption.

Comment (winsvega) [inline getcoverage.sh]: this script is running from cpp-ethereum dir. what assumption it should be?

Comment (chfast) [inline getcoverage.sh]: I don't have a build in "build" folder. How can I use this script then?

**Comment by chfast** [comment]
Still not 100% happy about the script but let it land.
